### ISSUE TITLE
Not a structure, FieldChain (check_syntax) 

### ISSUE THREAD

**Issue by ed-holland** [abaplint/abaplint#1250]
Issue:
```
./src/nw/#jcs#bc_job_abap_debugging.prog.abap[418, 3]                      - Not a structure, FieldChain (check_syntax)
```

Source:
![image](https://user-images.githubusercontent.com/53995991/91588162-0811ce80-e958-11ea-8c8d-9a4a79e72aaf.png)

Definitions are in include:
![image](https://user-images.githubusercontent.com/53995991/91588127-fa5c4900-e957-11ea-80bb-3e435ec44fb6.png)

Playground does not complain if the definition is inside the same file.

**Comment by larshp** [comment]
hmm, I cannot reproduce this

I'm guessing it has something to do with `text-030`, can you try deleting line 420? and see if the error is still there?

**Comment by ed-holland** [comment]
You could be right.. there are other errors as well
```
./src/nw/#jcs#bc_job_abap_debugging.prog.abap[418, 3]                      - Not a structure, FieldChain (check_syntax)
./src/nw/#jcs#bc_job_abap_debugging.prog.abap[446, 3]                      - Not a structure, FieldChain (check_syntax)
./src/nw/#jcs#bc_job_abap_debugging.prog.abap[453, 3]                      - Not a structure, FieldChain (check_syntax)
./src/nw/#jcs#bc_job_abap_debugging.prog.abap[460, 3]                      - Not a structure, FieldChain (check_syntax)
./src/nw/#jcs#bc_job_abap_debugging.prog.abap[467, 3]                      - Not a structure, FieldChain (check_syntax)
./src/nw/#jcs#bc_job_abap_debugging.prog.abap[475, 3]                      - Not a structure, FieldChain (check_syntax)
./src/nw/#jcs#bc_job_abap_debugging.prog.abap[483, 3]                      - Not a structure, FieldChain (check_syntax)
./src/nw/#jcs#bc_job_abap_debugging.prog.abap[491, 3]                      - Not a structure, FieldChain (check_syntax)
./src/nw/#jcs#bc_job_abap_debugging.prog.abap[504, 3]                      - Not a structure, FieldChain (check_syntax)
``

446: `  l_fcat-seltext_l = text-010.`
453: `l_fcat-seltext_l = text-011.`
...

They are all references to text-*.

**Comment by larshp** [comment]
is `bc_job_abap_debugging.prog.abap` an include or something special?

**Comment by ed-holland** [comment]
It is a report. Nothing special, just a tool to diagnose issues @ customer

**Comment by larshp** [comment]
hmm, any shadowing? eg. variables or types somewhere that redefine the `text` variable, like

`DATA text TYPE c LENGTH 1.`

**Comment by larshp** [comment]
yea, looks like its the issue, reproduced as:

```abap
DATA text TYPE c LENGTH 1.
WRITE TEXT-001.
```

**Comment by larshp** [comment]
ref https://github.com/abaplint/abaplint/pull/1251/commits/e82e598c5ae4a2e9245b8f4e39a4a5983148bcf1

**Comment by larshp** [comment]
try again, 2.39.23 published

**Comment by ed-holland** [comment]
Confirmed, there is a variable text :-)
![image](https://user-images.githubusercontent.com/53995991/91653231-f4bd4b00-ea9e-11ea-86ec-555c017614dc.png)

**Comment by ed-holland** [comment]
Issue resolved. Thanks
